ui/app/pages/settings/networks-tab/network-form/network-form.component.js

```diff
@@ -294,7 +315,18 @@ export default class NetworkForm extends PureComponent {
 
     if (providerError || typeof endpointChainId !== 'string') {
       errorMessage = t('failedToFetchChainId')
-    } else if (chainId !== endpointChainId) {
+    } else if (parsedChainId !== endpointChainId) {
+      if (!formChainId.startsWith('0x')) {
+        try {
+          endpointChainId = new BigNumber(endpointChainId, 16).toString(10)
+        } catch (err) {
+          log.error(
+            'Failed to convert endpoint chain ID to decimal',
```

Comment:
Should this case result in an error being shown to the user as well? Presumably if we get here, there has either been a network error or the network is returning an invalid value for eth_chainId. Should the user know that?

Reply:
Oh, well maybe this will be covered by the existing error message below

Reply:
Yeah, that's the idea!